src/CattyTests/Views/Data/BrickCellVariableDataTests.swift

```diff
@@ -0,0 +1,164 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+class BrickCellVariableDataTests: XCTestCase {
+
+    func testInitWithFrameWhenObjectVariableInContainer() {
+
+        let spriteObject = SpriteObject()
+        spriteObject.name = "testObject"
+
+        let variablesContainer = VariablesContainer()
+        let userVariable1 = UserVariable(name: "testVariable1")
+        let userVariable2 = UserVariable(name: "testVariable2")
+        let userVariable3 = UserVariable(name: "testVariable3")
+        variablesContainer.addObjectVariable(userVariable1, for: spriteObject)
+        variablesContainer.addObjectVariable(userVariable2, for: spriteObject)
+        variablesContainer.programVariableList.add(userVariable3 as Any)
+
+        let project = Project()
+        project.variables = variablesContainer
+
+        spriteObject.project = project
+
+        let script = Script()
+        script.object = spriteObject
+
+        let brick = ChangeVariableBrick()
+        brick.userVariable = userVariable2
+        brick.script = script
+
+        let brickCell = ChangeVariableBrickCell()
+        brickCell.scriptOrBrick = brick
+
+        let brickCellVariableData = BrickCellVariableData(frame: CGRect(), andBrickCell: brickCell, andLineNumber: 0, andParameterNumber: 0)
+
+        XCTAssertTrue(brickCellVariableData?.currentValue == brick.userVariable.name)
+
+        var allVariableNames = [String]()
+        if let allVariables = variablesContainer.allVariables(for: spriteObject) as? [UserVariable] {
+            allVariableNames.append(kLocalizedNewElement)
+            for variable in allVariables {
+                allVariableNames.append(variable.name)
+            }
+            allVariableNames.sort()
+        }
+        let values = (brickCellVariableData?.values as! [String]).sorted()
```

Comment:
Please remove the `sorted` here. The order plays a critical role.

Reply:
I have now removed `sorted` and as per `allVariables` method of `VariablesContainer`, I have assumed that program variables will be listed first, then the object variables and then other variables which aren't in the VariablesContainer. So what I understand from this is: this order needs to be maintained also when refactoring is done to these methods in future. Please correct me if I misunderstood.